### PR TITLE
Copy test s2s secret

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[AKS tests - notification service](https://tools.hmcts.net/jira/browse/BPS-1233)

### Change description ###

for aks tests this has to be within same resource group

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -102,4 +107,19 @@ resource "azurerm_key_vault_secret" "staging_db_database" {
...
+# endregion
+
+# region Copy secrets from BulkScan
+
+data "azurerm_key_vault_secret" "bulk_scan_processor_s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.bulk_scan_key_vault.id}"
+  name         = "s2s-secret"
+}
+
+resource "azurerm_key_vault_secret" "bulk_scan_processor_s2s_secret_copy" {
```
Comment (nitinprabhu) [inline infrastructure/main.tf:119]: can be bulk_scan_processor_s2s_secret_copy  -> test-bsp-s2s-secret as we do in other repos but not too strong about it

File: infrastructure/main.tf [hunk 2]
```diff
@@ -102,4 +107,19 @@ resource "azurerm_key_vault_secret" "staging_db_database" {
...
+data "azurerm_key_vault_secret" "bulk_scan_processor_s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.bulk_scan_key_vault.id}"
+  name         = "s2s-secret"
+}
+
+resource "azurerm_key_vault_secret" "test_bsp_s2s_secret" {
+  key_vault_id = data.azurerm_key_vault.reform_scan_key_vault.id
+  name         = "test-bsp-s2s-secret"
+  value        = data.azurerm_key_vault_secret.bulk_scan_processor_s2s_secret.value
+}
```
Comment (lgonczar) [inline infrastructure/main.tf:123]: How about just test-s2s-secret? All we care about is that it's an s2s secret for tests.

Maybe we could also create a pair of secrets in s2s (microservice key and name) and use those from s2s key vault?

File: infrastructure/main.tf [hunk 3]
```diff
@@ -102,4 +107,19 @@ resource "azurerm_key_vault_secret" "staging_db_database" {
...
 
-# endregion
\ No newline at end of file
+# endregion
+
+# region Copy secrets from BulkScan
+
+data "azurerm_key_vault_secret" "bulk_scan_processor_s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.bulk_scan_key_vault.id}"
+  name         = "s2s-secret"
```
Comment (lgonczar) [inline infrastructure/main.tf:116]: If we copy the secret, it should be test-s2s-secret. But my first suggestion is defining a separate secret pair in s2s, as it's probably very little work.

Comment (doncem) [reply infrastructure/main.tf:116]: defining lot's of secrets just for tests is not recommended and i did mention that a while back ago when writing those functional tests for the service.

should be re-used. and yes, good notice - i should re-use the test one from bsp

File: infrastructure/main.tf (hunk 2 above)
Comment (doncem) [reply infrastructure/main.tf:123]: well, it's a different resource group. there's not harm in having extra prefix level

File: infrastructure/main.tf (hunk 3 above)
Comment (lgonczar) [reply infrastructure/main.tf]: I don't know, separate test microservice secrets for tests would make a lot of sense, given this is a separate product

File: infrastructure/main.tf (hunk 2 above)
Comment (lgonczar) [reply infrastructure/main.tf]: It's a bit confusing to have a secret named "test bulk scan processor's s2s secret" in reform scan notification service, which itself is a part of a different product and is unaware of bulk scan processor.

**Comment by doncem** [comment]
Merging without original approval. Implementation is as expected by the team (members) and should unblock further development

**Pull request merged**